spectra/nv_ground_state.py

```diff
@@ -216,14 +276,107 @@ def plot_nv_ground_magnetic_transition_amplitudes(transition_bvec, static_bvec,
+    fixed.
+    :param frequencies: Array of frequencies over which to calculate spectrum
+    :param transition_bvec: The magnetic field vector for the magnetic transition (RF field)
+    :param static_bvec: The static bias field that determines the eigenstates
+    :initial_populations: Array of initial populations in eigen-energy order from ground state to highest energy
+    :param p: Parameters describing the NV center
+    :return: returns the probability of a transition occuring for a particular driving RF field.
+    """
+    hh_int = get_magnetic_transition_operator(p, transition_bvec)
+    energies, eigenstates = get_nv_ground_eigenspectrum(p, static_bvec)
+
+    transition_energies, transition_amplitudes = get_transition_amplitudes(hh_int, energies, eigenstates,
+                                                                           initial_populations=initial_populations)
+
+    spectrum = np.zeros_like(frequencies)
+    for i, amp in enumerate(transition_amplitudes):
+        spectrum += lorentz_lineshape(frequencies, transition_energies[i], amp)
+
+    return spectrum
+
+def get_power_broadened_spectrum_nv_axis_average(frequencies, transition_bvec, static_bvec, initial_state,
+                                  p: NVGroundParameters14N=NVGroundParameters14N(),
+                                                 nv_axes=((1,1,1), (-1,-1,1), (1,-1,-1), (-1,1,-1))):
+    """
+    Gets the spectral average over all possible NV axis directions (legs of tetrahedron)
+    Args:
+        frequencies: line space for frequencies
+        transition_bvec: B-field at max amplitude for oscillating RF wave
+        static_bvec: static applied B-field
+        initial_state: distribution of population among ground eigenstates
+        p: NV ground parameters
+
+    Returns: spectrum to be plotted
+
+    """
+    bvecs = get_bfields(static_bvec, nv_axes[0], nv_axes[1], nv_axes[2], nv_axes[3])
+    transition_bvecs = get_bfields(transition_bvec, nv_axes[0], nv_axes[1], nv_axes[2], nv_axes[3])
+
+    spectrum = np.zeros_like(frequencies)
+    for i, bvec in enumerate(bvecs):
+        spectrum += get_power_broadened_spectrum(frequencies, np.array(transition_bvecs[i]), bvec,
+                                                 initial_populations=initial_state)
+    return spectrum / len(bvecs)
+
 
 if __name__ == "__main__":
-    bmag = 4.6E-3
-    phi = 0. * np.pi / 180. # Polar angle
-    theta = 0.
-    static_bvec = bmag * np.array([np.sin(phi) * np.cos(theta), np.sin(phi) * np.sin(theta), np.cos(phi)])
-    transition_bvec = np.array([0., 1.E-4, 0.E-4])
-
-    plot_nv_ground_magnetic_transition_amplitudes(transition_bvec=transition_bvec,
-                                                  static_bvec=static_bvec)
-    plot_nv_ground_eigenspectrum(NVGroundParameters14N(), bvector=static_bvec)
+
+# ----------------------------------------------------------------------------------------------------------------------
```

Comment:
Possibly more Pythonic to use the set of three quotation marks instead when commenting multiple line

Reply:
Got rid of the other examples and thus anything commented in the main.